ui/packages/catalog-ui-search/src/main/webapp/component/query-time/query-time.view.js

```diff
@@ -120,6 +107,29 @@ module.exports = Marionette.LayoutView.extend({
         this.$el.toggleClass('is-timeRange-between', timeRange === 'between');
         this.$el.toggleClass('is-timeRange-relative', timeRange === 'relative');
     },
+    setupTemporalSelections: function () {
+        var metacards = metacardDefinitions.sortedMetacardTypes.filter((metacard) => metacard.type === "DATE");
+        var definitions = [];
+        for(var i = 0; i < metacards.length; i++) {
+            var metacard = metacards[i];
+            definitions.push({
+                label: metacard.id,
+                value: metacard.id
+            });
+        }
```

Comment:
You could replace this with a map function. (Someone might be able to reduce this even further)

```javascript
    var definitions = metacards.map(metacard => {
        return {
            label: metacard.id, 
            value: metacard.id
        }
    })
```

Reply:
```javascript
    var definitions = metacards.map(metacard => ({label: metacard.id, value: metacard.id}))
``` Will this work?